community/index/src/main/java/org/neo4j/index/gbptree/RightmostInChain.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.index.gbptree;
+
+import org.neo4j.io.pagecache.PageCursor;
+
+import static org.neo4j.index.gbptree.TreeNode.NO_NODE_FLAG;
+
+/**
+ * Used to verify a chain of siblings starting with leftmost node.
+ * Call {@link #assertNext(PageCursor)} with cursor pointing at sibling expected to be right sibling to previous call
+ * to verify that they are indeed linked together correctly.
+ * <p>
+ * When assertNext has been called on node that is expected to be last in chain, use {@link #assertLast()} to verify.
+ */
+class RightmostInChain<KEY>
+{
+    private final TreeNode<KEY,?> node;
+    private final int stableGeneration;
+    private final int unstableGeneration;
+
+    private long currentRightmost;
+    private long expectedNextRightmost;
+
+    RightmostInChain( TreeNode<KEY,?> node, int stableGeneration, int unstableGeneration )
+    {
+        this.node = node;
+        this.stableGeneration = stableGeneration;
+        this.unstableGeneration = unstableGeneration;
+        this.currentRightmost = TreeNode.NO_NODE_FLAG;
+        this.expectedNextRightmost = TreeNode.NO_NODE_FLAG;
+    }
+
+    long assertNext( PageCursor cursor )
+    {
+        long pageId = cursor.getCurrentPageId();
+        long leftSibling = node.leftSibling( cursor, stableGeneration, unstableGeneration );
+        long rightSibling = node.rightSibling( cursor, stableGeneration, unstableGeneration );
+
+        // Assert we have reached expected node and that we agree about being siblings
+        assert leftSibling == currentRightmost :
```

Comment:
Yes, this should be fixed. Right now only used in tests and we are not sure how we want it to look, therefore we suggest postponing changing this.